tests/primer/primer_tool.py

```diff
@@ -8,6 +8,7 @@
 import json
 from io import StringIO
 from pathlib import Path
+from typing import Dict, List, Union
```

Comment:
any reason not to do __future__ style types?

Reply:
`PackageMessages` is created at runtime so we need the `typing` imports.

`if TYPE_CHECKING` could also work, but so far our "style" has been to keep typing variables in the main scope and just use `from typing` imports.